aqi_app/src/main.ts

```diff
@@ -0,0 +1,359 @@
+import * as url from 'url';
+
+import { https } from 'follow-redirects';
+import { PainterOptions, Painter, Frame, ResourceManager } from 'camstreamerlib/CamOverlayPainter/Painter';
+
+type ImageCode = {
+    text: string[];
+    img_file: string;
+    color: [number, number, number];
+};
+type AqiResponseType = {
+    data: {
+        aqi: number;
+    };
+};
+
+let settings: {
+    camera_protocol: string;
+    camera_user: string;
+    camera_pass: string;
+    camera_ip: string;
+    camera_port: number;
+    scale: number;
+    coordinates: string;
+    access_token: string;
+    display_location: string;
+    location: string;
+    update_frequency: number;
+    pos_x: number;
+    pos_y: number;
+    resolution: string;
+
+    serviceName: string;
+    font: string;
+    translation: Record<string, string>;
+};
+
+let cam_width: number;
+let cam_height: number;
+
+let background: Painter;
+let value: Frame;
+let label: Frame;
+let text: Frame;
+let upperText: Frame;
+let lowerText: Frame;
+
+const codes: Record<string, ImageCode> = {
+    good: {
+        text: ['GOOD'],
+        img_file: 'Good.png',
+        color: [0, 153 / 255, 76 / 255],
+    },
+    moderate: {
+        text: ['MODERATE'],
+        img_file: 'Moderate.png',
+        color: [1.0, 1.0, 51 / 255],
+    },
+    sensitive: {
+        text: ['UNHEALTHY FOR', 'SENSITIVE GROUPS'],
+        img_file: 'sensitive_groups.png',
+        color: [1.0, 128 / 255, 0],
+    },
+    unhealthy: {
+        text: ['UNHEALTHY'],
+        img_file: 'Unhealthy.png',
+        color: [1.0, 51 / 255, 51 / 255],
+    },
+    very_unhealthy: {
+        text: ['VERY UNHEALTHY'],
+        img_file: 'Very_Unhealthy.png',
+        color: [102 / 255, 0, 204 / 255],
+    },
+    hazardous: {
+        text: ['HAZARDOUS'],
+        img_file: 'Hazardous.png',
+        color: [153 / 255, 0, 0],
+    },
+    error: {
+        text: ['ERROR'],
+        img_file: 'Error.png',
+        color: [0, 0, 0],
+    },
+};
+
+function split(text: string): string[] {
+    if (text.length <= 19) {
+        return [text];
+    }
+
+    const half = text.length / 2;
+    let middle_space = -1;
+    for (let f = 0; f < text.length; f += 1) {
+        if (text[f] === ' ') {
+            if (Math.abs(half - f) < Math.abs(half - middle_space)) {
+                middle_space = f;
+            } else {
+                break;
+            }
+        }
+    }
+    if (middle_space === -1) {
+        return [text];
+    } else {
+        return [text.substring(0, middle_space), text.substring(middle_space + 1)];
+    }
+}
+
+function setCodeText() {
+    settings.translation ??= {};
+    for (const c in codes) {
+        if (settings.translation[c]) {
+            codes[c].text = split(settings.translation[c]);
+        }
+    }
+}
+
+function registerResources() {
+    const rm = new ResourceManager();
+    for (const c in codes) {
+        rm.registerImage(codes[c].img_file, `${settings.serviceName}/${codes[c].img_file}`);
```

Comment:
Mame ulozene obrazky pro EN a Japonstinu, ale co kdyz si do prekladu vlozim uplne jiny jazyk? Podle kodu si asi musis vybrat EN, je to tak v poradku?

Reply:
Do překladu si můžeš vložit, co tě zrovna napadne. Pozadí a font se nastavuje nezávisle na překladech, tedy kombinace (český překlad, anglické pozadí, japonský font) je povolená. Stejně tak třeba (arabský překlad, japonské pozadí, anglický font) - i když se to asi nezobrazí správně.
Každopádně pro pozadí máš právě dvě možnosti, webové rozhraní ti ani nedává na výběr. Pokud bys měl nějakou jinou představu, můžu to předělat, ale mě to takhle dává smysl.